common/spec/dependabot/dependency_group_spec.rb

```diff
@@ -1,14 +1,105 @@
 # frozen_string_literal: true
 
 require "dependabot/dependency_group"
+require "dependabot/dependency"
 
+# TODO: Once the Updater has been merged into Core, we should test this
+# using the DependencyGroupEngine methods instead of mocking the functionality
 RSpec.describe Dependabot::DependencyGroup do
+  let(:dependency_group) { described_class.new(name, rules) }
+  let(:name) { "test_group" }
+  let(:rules) { ["test-*"] }
+
+  let(:test_dependency_1) do
+    Dependabot::Dependency.new(
+      name: "test-dependency-1",
+      package_manager: "bundler",
+      version: "1.1.0",
+      requirements: [
+        {
+          file: "Gemfile",
+          requirement: "~> 1.1.0",
+          groups: [],
+          source: nil
+        }
+      ]
+    )
+  end
+
+  let(:test_dependency_2) do
+    Dependabot::Dependency.new(
+      name: "another-test-dependency",
+      package_manager: "bundler",
+      version: "1.1.0",
+      requirements: [
+        {
+          file: "Gemfile",
+          requirement: "~> 1.1.0",
+          groups: [],
+          source: nil
+        }
+      ]
+    )
+  end
+
   describe "#name" do
     it "returns the name" do
-      my_dependency_group_name = "Darren from work"
-      dependency_group = described_class.new(my_dependency_group_name)
+      expect(dependency_group.name).to eq(name)
+    end
+  end
+
+  describe "#rules" do
+    it "returns a list of rules" do
+      expect(dependency_group.rules).to eq(rules)
+    end
+  end
+
+  describe "#dependencies" do
+    context "when calculate_dependency_groups! has not been run" do
+      it "returns an empty list" do
+        expect(dependency_group.dependencies).to eq([])
+      end
+    end
```

Comment:
Not blocking, I think we can avoid drawing in the language of the `DependencyGroupEngine` class and just test this was 
```suggestion
    context "when no dependencies are assigned to the group" do
      it "returns an empty list" do
        expect(dependency_group.dependencies).to eq([])
      end
    end
```